dotnet/src/SemanticKernel.Core/Functions/DefaultKernelPlugin.cs

```diff
@@ -30,7 +30,9 @@ internal DefaultKernelPlugin(string name, string? description, IEnumerable<Kerne
             foreach (KernelFunction f in functions)
             {
                 Verify.NotNull(f, nameof(functions));
-                this._functions.Add(f.Name, f);
+
+                var cloned = f.CreateForPlugin(name);
+                this._functions.Add(cloned.Name, cloned);
```

Comment:
If `cloned.Name` is always the same `name` you can avoid an extra allocation of `cloned`.

```suggestion
                this._functions.Add(name, f.CreateForPlugin(name));
```

Reply:
`cloned.Name` is the name of the function and `name` is the name of the plugin